### ISSUE TITLE
Crash: compact formatter with any fractional part

### ISSUE THREAD

**Issue by chris-rutkowski** [a14n/dart-decimal#107]
Hello.

```
flutter: 3.22.1
decimal: 3.0.1
intl: 0.19.0
```

Compact formatter works good for integer decimal values, like here

```
final result = DecimalFormatter(NumberFormat.compact()).format(Decimal.parse('1000.0'))
result == "1K"
```

But:

```
final result = DecimalFormatter(NumberFormat.compact()).format(Decimal.parse('1000.1'))

Unhandled exception:
type '_DecimalIntl' is not a subtype of type 'num' of 'other'
#0      _CompactNumberFormat._divide (package:intl/src/intl/compact_number_format.dart:470:52)
#1      _CompactNumberFormat._styleFor (package:intl/src/intl/compact_number_format.dart:519:34)
#2      _CompactNumberFormat.format (package:intl/src/intl/compact_number_format.dart:424:17)
#3      DecimalFormatter.format (package:decimal/intl.dart:38:49)
#4      Eval ()
...
```

For comparison standard parser

```
final result1 = NumberFormat.compact().format(1000.0)
result1 == "1K"

final result2 = NumberFormat.compact().format(1000.1)
result2 == "1K"
```

let me know if I can be of any more help.

**Comment by a14n** [comment]
Thanks for the report. This should be fixed via 5f1bf8b7a01e3a4ac950037734011eb5178272cd

**Comment by a14n** [comment]
Fix published in 3.0.2
